Point_set_processing_3/include/CGAL/grid_simplify_point_set.h

```diff
@@ -44,11 +45,26 @@ namespace CGAL {
 namespace internal {
 
 
-/// Utility class for grid_simplify_point_set():
-/// Less_epsilon_points_3 defines a 3D points order / 2 points are equal
-/// iff they belong to the same cell of a grid of cell size = epsilon.
+// Round number to multiples of epsilon
+inline int round_epsilon(double value, double epsilon)
+{
+  double x = value / epsilon;
+  if (x > 0.)
+    return static_cast<int> (x);
```

Comment:
Indeed… one first step would be to replace that by `ptrdiff_t` to have the largest range possible, but the problem remains. In practice, I doubt this will ever happen in real cases. Maybe a solution would be to add a fallback mode with a define CGAL_GRID_SIMPLIFY_USE_ROUND or something?